project/Build.scala

```diff
@@ -546,6 +554,24 @@ object Build {
       }
   )
 
+  object ProjectTaskKeyFilter {
+    private val ignoredStyleCheckProjects = Seq(
+      scalalib,
+      partest,
+    ).flatMap(_.componentProjects)
+  }
+
+  implicit class ProjectTaskKeyFilter(val p: Project) extends AnyVal {
+    def include(key: TaskKey[_]): Boolean = {
+      if (key == scalastyleCheck) {
+        val isIgnore = ProjectTaskKeyFilter.ignoredStyleCheckProjects.contains(p)
+        !isIgnore
+      } else {
+        true
+      }
+    }
+  }
```

Comment:
Instead of this, consider simply overriding `scalastyleCheck` in the projects where we do not want it.

Reply:
Updated, see if this approach matches what you are suggesting here